optimum/exporters/ipex/modeling/modeling_llama.py

```diff
@@ -0,0 +1,208 @@
+        **kwargs,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
+        """
+        Args:
+            hidden_states (`torch.FloatTensor`): input to the layer of shape `(batch, seq_len, embed_dim)`
+            attention_mask (`torch.FloatTensor`, *optional*):
+                attention mask of size `(batch_size, sequence_length)` if flash attention is used or `(batch_size, 1,
+                query_sequence_length, key_sequence_length)` if default attention is used.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+            use_cache (`bool`, *optional*):
+                If set to `True`, `past_key_values` key value states are returned and can be used to speed up decoding
+                (see `past_key_values`).
+            past_key_value (`Tuple(torch.FloatTensor)`, *optional*): cached past key and value projection states
+            residual (`torch.Tensor`): residual tensor to the layer of shape `
+        """
+
+        self.preprocess_for_optimize(hidden_states, past_key_value, kwargs)
+
+        query, key, value = self.qkv_gemm(hidden_states, kwargs)
+
+        key, value = self.rope(key, value, position_ids, past_key_value, kwargs)
+
+        present = self.get_present(query, key, value, use_cache)
+
+        attn_output, attn_weight = self.sdpa(query, key, value, attention_mask, past_key_value, kwargs)
+
+        attn_output = self.out_proj(attn_output, residual)
+
+        self.post_process_for_optimize()
+        outputs = (attn_output, present)
+        if output_attentions:
+            outputs += (attn_weight, )
+        else:
+            outputs += (None, )
+        return outputs
+
+class _IPEXLlamaMLP(nn.Module):
+    def __init__(self, module, config, distributed=False) -> None:
+        self.module = module
+        self.config = config
+        self.distributed = distributed
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        residual: torch.Tensor,
+        **kwargs
+    ):
+        """
+        Args:
+            hidden_states (`torch.FloatTensor`): input to the layer of shape `(batch, seq_len, embed_dim)`
+        """
+        pass
+
+
+class _IPEXLlamaDecoderLayer(nn.Module):
+    def __init__(self, module, config, distributed=False) -> None:
+        super().__init__()
+        self.layer_idx = module.self_attn.layer_idx
+        self.attn = _IPEXLlamaAttention(module.self_attn, config, distributed)
+        self.mlp = _IPEXLlamaMLP(module.mlp, config, distributed)
+        self.input_layernorm = ipex.llm.modules.RMSNorm(module.input_layernorm.weight, module.input_layernorm.variance_epsilon)
+        self.post_attention_layernorm = ipex.llm.modules.RMSNorm(module.post_attention_layernorm.weight, module.post_attention_layernorm.variance_epsilon)
```

Comment:
RMS norm was already in [here](https://github.com/huggingface/optimum-intel/blob/main/optimum/exporters/ipex/modeling_utils.py#L27-L29)